File: src/main/java/com/destino/projeto_destino/model/Viagem.java
package com.destino.projeto_destino.model;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "via_viagem")
public class Viagem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private List<String> itens;
    private double valor;

    public Viagem(String nome, List<String> itens, double valor) {
        this.nome = nome;
        this.itens = itens;
        this.valor = valor;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public List<String> getItens() {
        return itens;
    }

    public void setItens(List<String> itens) {
        this.itens = itens;
    }

    public String getValor() {
        return String.format("%.2f", valor);
    }

    public void setValor(double valor) {
        this.valor = valor;
    }
}

File: src/main/kotlin/com/fatec/destino/model/viagem/Viagem.kt
package com.fatec.destino.model.viagem

import com.fasterxml.jackson.annotation.JsonFormat
import com.fatec.destino.model.pacote.Pacote
import com.fatec.destino.util.model.pacote.PacoteStatus
import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "via_viagem")
class Viagem(
    @ManyToOne
    @JoinColumn(name = "PAC_ID", referencedColumnName = "PAC_ID", nullable = false)
    var pacote: Pacote,

    @JsonFormat(pattern = "dd-MM-yyyy")
    @Column(name = "VIA_DATA_INICIO", nullable = false)
    var inicio: LocalDate,

    @JsonFormat(pattern = "dd-MM-yyyy")
    @Column(name = "VIA_DATA_FIM", nullable = false)
    var fim: LocalDate,

    @Column(name = "VIA_DISPONIBILIDADE", nullable = false)
    var disponibilidade: Int = 0,

    @Column(name = "VIA_STATUS", nullable = false)
    @Enumerated(EnumType.STRING)
    var status: PacoteStatus = PacoteStatus.EMANDAMENTO
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "VIA_ID", nullable = false)
    var id: Long? = null

    fun recalcularStatus() {
        this.status = when {
            this.status == PacoteStatus.CANCELADO -> PacoteStatus.CANCELADO
            this.fim.isBefore(LocalDate.now()) -> PacoteStatus.CONCLUIDO
            else -> PacoteStatus.EMANDAMENTO
        }
    }
}
